app/api/shorten.py

```diff
@@ -0,0 +1,256 @@
+from voluptuous import MultipleInvalid
+
+
+dotenv.load()
+site_url = dotenv.get('SITE_URL')
+
+
+def check_authentication_with_token():
+    """Check that a user is authenticated and used token for authentication."""
+    if g.current_user.is_anonymous or not g.token_used:
+        abort(403)
+
+
+@api.route('/shorten', methods=['POST'], strict_slashes=False)
+@auth.login_required
+def shorten_url():
+    """An endpoint which shortens a long url.
+
+    It makes use of a vanity string if supplied by a registered user.
+    """
+    check_authentication_with_token()
+
+    try:
+        valid_url(request.json)
+    except MultipleInvalid:
+        abort(400, "Incomplete or too many required keys provided.")
+    except BadRequest:
+        abort(400, "The request does not contain a body.")
+
+    long_url = request.json.get('url')
+
+    if request.json.get('vanity_string') and not g.current_user.is_anonymous:
+        vanity_string = request.json.get('vanity_string')
+        short_url = UrlSaver.generate_and_save_urls(long_url, g.current_user,
+                                                    vanity_string)
+        return jsonify({'id': short_url.id,
+                        'short_url': site_url + short_url.short_url}), 201
+
+    user = g.current_user
+
+    short_url = UrlSaver.generate_and_save_urls(long_url, user)
+    return jsonify({'id': short_url.id,
+                    'short_url': site_url + short_url.short_url}), 201
+
+
+@api.route('/<shorturl>', strict_slashes=False)
+def get_url(shorturl):
+    """Redirect short_urls to their long_url version.
+
+    Given a short url, the code looks up the short url in database
+    and if found redirects to the long url path. if not found sends a 404
+    status_code.
+    """
+    short_url = ShortUrl.findurl(shorturl, request)
+    if short_url and short_url.deleted:
+        abort(404, "This URL has been deleted.")
+    elif short_url and not short_url.is_active:
+        abort(400, "URL is inactive.")
+    elif short_url and short_url.is_active:
+        return redirect(short_url.long_url.long_url, code=302)
+    else:
+        abort(404)
+
+
+@api.route('/short_url/<int:id>/change_longurl', methods=['PUT'],
+           strict_slashes=False)
+@auth.login_required
+def change_long_url(id):
+    """Update target url of a given short_url."""
+    check_authentication_with_token()
+
+    try:
+        valid_url(request.json)
+    except MultipleInvalid:
+        abort(400, "Incomplete or too many required keys provided.")
+    except BadRequest:
+        abort(400, "The request does not contain a body.")
+
+    try:
+        short_url = [short_url for short_url in g.current_user.short_urls
+                     if short_url.id == id and not short_url.deleted][0]
+    except IndexError:
+        abort(404, "You do not have any such URL.")
+
+    long_url = request.json.get('url')
+    result = short_url.change_long_url(long_url, g.current_user)
+    if result is True:
```

Comment:
According to [pep 8 programming recommendations](https://www.python.org/dev/peps/pep-0008/#programming-recommendations), 
```Don't compare boolean values to True or False using == .

Yes:   if greeting:
No:    if greeting == True:
Worse: if greeting is True:
```
In this case, `if result` is a better option to use.